ignite/engine/engine.py

```diff
@@ -1002,17 +1000,19 @@ def _internal_run_as_gen(self) -> Generator[Any, None, State]:
             time_taken = time.time() - start_time
             # time is available for handlers but must be updated after fire
             self.state.times[Events.COMPLETED.name] = time_taken
-            handlers_start_time = time.time()
 
-            # do not fire Events.COMPLETED if we terminated the run with flag `skip_completed_after_termination=True`
-            if self.should_terminate and not self.skip_completed_after_termination or not self.should_terminate:
+            if self.should_terminate and self.skip_completed_after_termination:
+                # do not fire Events.COMPLETED if we terminated the run with flag `skip_completed_after_termination=True`
+                hours, mins, secs = _to_hours_mins_secs(time_taken)
+                self.logger.info(f"Engine run terminated. Time taken: {hours:02d}:{mins:02d}:{secs:06.3f}")
```

Comment:
Let's refactor this to avoid code duplication and finally use `not (...)`

Reply:
Solved by using "finished" in the log message in place of "terminated" and "completed".